### ISSUE TITLE
[NFR] Snap-areas should not be visible when trigger is pointer and outside container

### ISSUE THREAD

**Issue by fq-selbach** [Flyer53/jsPanel4#83]
Sorry for spamming the issue section 🙈 but I have another one ^_^. This time it comes with a possible solution included ;-)

I noticed that the snap areas are still active when the mouse pointer is already outside the container (I get this a lot because I have containers-panels that are much smaller than the screen). This is due to the way the distance is calculated, it does not take into account negative values.

For `snap.trigger == "panel"` this makes sense and I'd even call it a feature not a bug, but for `snap.trigger == "pointer"` it is rather confusing, especially in my case where I use the overflow to move the panel from one container to another.

So here is what I'm suggesting for the dragElmt method (https://github.com/Flyer53/jsPanel4/blob/master/dist/jspanel.js#L1040):
```
var checkSnapArea = true;
if (opts.snap.trigger == "pointer"){
	if (elmtL < 0 || elmtR < 0 || elmtT < 0 || elmtB < 0){
		//pointer outside container
		checkSnapArea = false;
	}
}
if (checkSnapArea){
	if (lefttopVectorDrag < snapSens) {
		elmt.snappableTo = 'left-top';
...
```
This will check if the mouse pointer is inside the container and only then check for snap areas.

**Comment by Flyer53** [comment]
Ok, thanks for the input 😄 
I'll look at that ...

**Comment by Flyer53** [comment]
Mmmmh ... not sure about this 🤔 
I tend to think of it as feature in both cases. Although I never considered the drag and drop thing you obviously use.

If you have rather small containers and panels did you try reducing the default `sensitivity: 70` parameter to a lower value? Let's say to `sensitivity: 30`. Could improve it a bit for you.
And wouldn't it be better in your case to use the default `trigger: 'panel'`?

**Comment by fq-selbach** [comment]
I've been using `sensitivity: 40` already when I noticed that issue :-/
To be honest I'm not a big fan of `trigger: panel`, especially because it just prevents panels from being at a certain position (they will always snap when touching the area). `trigger: pointer` is pretty close to the behavior the users know from Windows and feel more natural to me.
I'll keep my own 'hack' for now until you've decided how (or if at all) you want to approach this issue ;-)

**Comment by Flyer53** [comment]
@fq-selbach 
I didn't look into this lately and there won't be any changes in the coming v4.7.0 probably.
But I consider a complete rewrite of the snap feature for v4.8.0

I'll post an update here as soon as I have something to test.

**Comment by Flyer53** [comment]
@fq-selbach 
Hi there,
You might have noticed that I released v4.8.0 recently. However, v4.8.0 does not implement a change of the snap feature.

But today I already uploaded [v4.9.0-alpha.1](https://alpha.jspanel.de/downloads/jspanel-4.9.0-alpha.1.zip) to [alpha.jspanel.de](https://alpha.jspanel.de/index.html)
v4.9.0-alpha.1 **includes a modification of the snap feature** and introduces a new snap parameter `active` which can be set to the string `"inside"`. **Together with parameter** `trigger: "pointer"` it results in a panel that will snap only if the mouse pointer is inside the panel's parent container.
The very last example in the [alpha dragit docs](https://alpha.jspanel.de/api.html#options/dragit) shows a working example. Otherwise there is nothing explanatory in the docs yet.

I would highly appriciate if you gave it a try ...

Regards,
Stefan

**Comment by fq-selbach** [comment]
Hi Stefan,

the demo works fine 👍 
As mentioned in the other issue I cannot upgrade easily right now before solving the version conflicts (original 4.6.0 and our mod. version) but this looks like I could remove our own workaround for this issue :-)

**Comment by Flyer53** [comment]
@fq-selbach 
Yep ... my same answer as to #82 😏

**Comment by Flyer53** [comment]
Updated to https://alpha.jspanel.de/downloads/jspanel-4.9.0-alpha.8.zip

**Comment by Flyer53** [comment]
Implemented in v4.9.0 release for panels appended to a container other than `<body>`
